code/java/FlintSources/src/main/kotlin/org/discipl/flint/sources/transformers/textline/QuintorTextLineTransformer.kt

```diff
@@ -0,0 +1,102 @@
+package org.discipl.flint.sources.transformers.textline
+
+import mu.KLogging
+import org.discipl.flint.sources.models.parts.ContainerImpl
+import org.discipl.flint.sources.models.parts.TextLineImpl
+import org.discipl.flint.sources.clients.nsx.QuintorApiNsxTextLineClient.QuintorApiNsxTextLine
+import org.discipl.flint.sources.models.parts.*
+
+class QuintorTextLineTransformer : NewTextLineTransformer<QuintorApiNsxTextLine> {
+    companion object : KLogging()
+
+    override fun transform(list: List<QuintorApiNsxTextLine>): List<SourcePart> {
+        val root = list.first { it.parent == null && it.next == null }
+        logger.info { "Root: $root" }
+        val remainingLines = ArrayList(list)
+        remainingLines.remove(root)
+        val transformedRoot = transform(root, remainingLines, list)
+        val transformedRootContainer = transformedRoot as Container
+        val result = (transformedRootContainer as? Container)?.children ?: emptyList()
+        if (remainingLines.isNotEmpty()) {
+            remainingLines.forEach { logger.warn { "We missed this line $it" } }
+        }
+        return result
+    }
+
+    private fun transform(
+        textLine: QuintorApiNsxTextLine,
+        remainingLines: MutableList<QuintorApiNsxTextLine>,
+        allLines: List<QuintorApiNsxTextLine>
+    ): SourcePart {
+        return when (textLine.textNodeType) {
+            "SimpleLine" -> return TextLineImpl(
+                textLine.id,
+                textLine.tag,
+                allLines.indexOf(textLine),
+                textLine.text!!
```

Comment:
`CsvTextlLineTransformer` and `JuriDecomposeTextLineTransformer` both have extra checks to ensure `text` and `prefix` fields are actually not null. Since `QuintorApiNsxTextLine` also defines them als nullable maybe an extra check is warranted or are we 100% sure that the `textNodeType` takes care of this? If this is case shouldn't `QuintorApiNsxTextLine` define them als not nullable?

Reply:
if the textNodeType is SimpleLine then the node is guaranteed to have text